### DISCUSSION TITLE
Upgrading versión of primevue 3.38 to 3.47.2, the theme fail.

### DISCUSSION THREAD

**Discussion by lpalacecr** [primefaces/community#1023] [PrimeVue]
Upgrade version of primevue 3.38 to 3.47.2, the theme failed.
I bought the Apollo theme version 5.0, I made a new project and when I updated the version of Primevue from 3.38 to 3.47.2 the inputSwitch stopped working, the checkbox, multiselect and radioButton display a circle when an option is marked.

I have seen that several bugs have been fixed from version 3.38 to 3.47.2, so I would like to be able to update the version of the apollo theme.

I attach a word document with several errors found so far.

Is it possible to update the Primevue version of a theme or do I have to stay with the version I have?

I appreciate the help you can give me.

Translated by Google Translate.

Original text:
Yo compré el tema apollo versión 5.0, hice un nuevo proyecto y cuando le actualicé la versión de primevue de la 3.38 a la 3.47.2 el inputSwitch dejó de funcionar, los checkbox,  multiselect  y lo radioButton desliegan un círculo caundo se marca una opción. 

He visto que de la versión 3.38 a la 3.47.2 se han corregido varios errores por lo que me gustaría poder actualizar la versión del tema apollo.

Adjunto documento en word con varios errores hasta ahora encontrados.

Es posible actualizar la versión de primevue de un tema o me tengo que quedar con la versión que tengo?

Agradezco la ayuda que me puedan brindar.

![imagen](https://github.com/primefaces/community/assets/2246195/fb9f5af2-787f-4fe5-b79d-37ddbeeab5aa)
[Upgrade Apollo theme.docx](https://github.com/primefaces/community/files/14058161/Upgrade.Apollo.theme.docx)

**Comment by Y0niix** [comment]
Same problem here with Freya theme.

**Reply by lpalacecr** [reply]
Hi, I think that updates should consider paid templates, because in each update not only come improvements, they also come with bug fixes, if they were just improvements there would be no problem in staying with the version when it was purchased, but if bug fixes come, it is Obviously the template must be updated to correct these errors, in my opinion.

PS: There are more errors, the ones shown are the ones I found in a quick review I did, since due to time reasons I have not been able to delve deeper into the review, because there are places where the template stops working completely.

Translated by Google Translate.

Hi, Pienso que las actualizaciones deberían considerar los templates de paga, porque en cada actualización no solo vienen mejoras tambien vienen correcciones a errores, si fueran solo mejoras no habría problema en quedarse con la versión cuando se compró, pero si vienen correcciones a errores es obvio que hay que actualizar el template para que se corrijan esos errores, es mi parecer.

PD: Hay más errores, los mostrados son los que encontré en una revisión rápida que hice, ya que por cuestiones de tiempo no he podido ahondar más en la revisión, porque hay lugares en que el template deja de funcionar totalmente.

Traducido por el traductor de google.

**Reply by Y0niix** [reply]
Well, even with updating the Freya theme template with last version, I have those bugs with Primevue 3.47.2. 
I had to downgrade to 3.46.0.

**Reply by malacay1462** [reply]
Yeah, i'm having the same error with Apollo..
Changing the version to 3.46.0 works - but that souldn't be the solution.. But at least it works with this version 🤷‍♂️

**Reply by lpalacecr** [reply]
Version 3.46 also has bugs, but it is functional and can be corrected in some way, but version 3.47.2 definitely crashes the application and I don't see a way to fix it. It seems that the problem is in the CSS, because if I remove it it works, but it remains without style, that is, the html checkbox.

La versión 3.46 también presenta bugs, pero queda funcional y se puede corregir de alguna manera, pero la versión 3.47.2 definitivamente se cae la aplicación y no veo una forma de arreglarlo. Parece que el problema está en el css, porque si lo quito funciona, pero queda sin estilo, es decir, el checkbox de html.

**Reply by pkulaszka** [reply]
Same issue with Sapphire theme. The last working version is 3.46.0

**Comment by basitiyidir** [comment]
The following command will fix it.

`sass --update public/layout/:public/layout/`

You may need to update your Sass version -> https://sass-lang.com/documentation/breaking-changes/slash-div/

**Reply by Y0niix** [reply]
Not working with last Freya theme 5.1.0, Primevue 3.47.2, and Sass 1.70.0

**Reply by basitiyidir** [reply]
I have Apollo and Ultimate and it worked.

I give this command in your theme root directory. Then everything works properly.

Sass v 1.69

`sass --update public/layout/:public/layout/`

**Reply by malacay1462** [reply]
Thanks for trying to help! Unfortunately, it didn't work for me with Apollo either (tried with sass 1.70.0 and 1.69.7 - Apollo v5). 
I also tried it with a new (clean) download of the Apollo theme (no custom changes, only updated package deps to newest version).

Also tried removing the already compiled `*.css` files before running the command, just in case it would change anything.

```bash
[2024-01-31 14:17] Compiled public/layout/styles/preloading/preloading.scss to public/layout/styles/preloading/preloading.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/blue/theme.scss to public/layout/styles/theme/theme-dark/blue/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/cyan/theme.scss to public/layout/styles/theme/theme-dark/cyan/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/green/theme.scss to public/layout/styles/theme/theme-dark/green/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/indigo/theme.scss to public/layout/styles/theme/theme-dark/indigo/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/orange/theme.scss to public/layout/styles/theme/theme-dark/orange/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/pink/theme.scss to public/layout/styles/theme/theme-dark/pink/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/purple/theme.scss to public/layout/styles/theme/theme-dark/purple/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dark/teal/theme.scss to public/layout/styles/theme/theme-dark/teal/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/blue/theme.scss to public/layout/styles/theme/theme-dim/blue/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/cyan/theme.scss to public/layout/styles/theme/theme-dim/cyan/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/green/theme.scss to public/layout/styles/theme/theme-dim/green/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/indigo/theme.scss to public/layout/styles/theme/theme-dim/indigo/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/orange/theme.scss to public/layout/styles/theme/theme-dim/orange/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/pink/theme.scss to public/layout/styles/theme/theme-dim/pink/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/purple/theme.scss to public/layout/styles/theme/theme-dim/purple/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-dim/teal/theme.scss to public/layout/styles/theme/theme-dim/teal/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/blue/theme.scss to public/layout/styles/theme/theme-light/blue/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/cyan/theme.scss to public/layout/styles/theme/theme-light/cyan/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/green/theme.scss to public/layout/styles/theme/theme-light/green/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/indigo/theme.scss to public/layout/styles/theme/theme-light/indigo/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/orange/theme.scss to public/layout/styles/theme/theme-light/orange/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/pink/theme.scss to public/layout/styles/theme/theme-light/pink/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/purple/theme.scss to public/layout/styles/theme/theme-light/purple/theme.css.
[2024-01-31 14:17] Compiled public/layout/styles/theme/theme-light/teal/theme.scss to public/layout/styles/theme/theme-light/teal/theme.css.
```

It was executed successfully, but unfortunately RadioButton/Checkbox/etc. are still broken.

But I just tried something different - to include the predefined PrimeVue component styles in addition. And the funny thing is - it has given better results (tested with `aura-dark-indigo`).

My way:
- Copy one of the predefined component themes from `node_modules/primevue/resources/themes` to `public/themes/` (for example `aura-dark-indigo`)
- Insert it into the "index.html" file before the template themes (if it is used afterwards, the styles of the template themes will be overridden)

```html
    <link rel="stylesheet" href="/themes/aura-dark-indigo/theme.css">
    <link href="/layout/styles/preloading/preloading.css" rel="stylesheet"/>
    <link id="theme-link" href="/layout/styles/theme/theme-dark/indigo/theme.css" rel="stylesheet"/>
```

This leads to better functioning elements - but not 100% (quick test: the radio button component does not display the active state correctly, but only when hovering).
This is not really a solution, but it kind of gives me the feeling that the styles of the (Apollo) theme for v3.47 are incomplete?

**Reply by lpalacecr** [reply]
Hi,
I replicated what was recommended and it worked perfectly for any component that has a checkbox, however, components still continue to fail, for example, the blog page continues to give errors, but without reviewing much, it seems that the datatable was removed from several slots that the theme uses, that is, it has nothing to do with the checkbox, it is another type of error.

Translated by Google Translate.

Original text:

Hi,
Repliqué lo recomendado y si funcionó perfectamente para cualquier componente que tenga un checkbox, sin embargo, todavía siguen fallando componentes, por ejemplo, la página de blog sigue dando error, pero sin revisar mucho, parece que al datatable le quitaron varios slots que el tema usa, es decir, no tiene que ver nada con el checkbox, es otro tipo de error.

**Reply by basitiyidir** [reply]
I checked. You are right.

There's nothing we can do but wait for the Prime team to intervene and find a solution to the problem and for the templates to be updated.

**Reply by malacay1462** [reply]
Yeah, I think the checkbox errors are related to missing css styles - thats why the use of component styles seems to be working (with a few problems like missing active state). 

For Blog Page > DataView, it seems there were breaking changes in 3.41 and the themes are build on an older version. https://github.com/primefaces/primevue/blob/master/CHANGELOG.md#3410-2023-11-20

Without much testing, I tried the following with seems to work for the component:

Change
```vue
<template #grid="{ data }"> 
	<div class="col-12 md:col-4">
		....
	</div>
</template>
```

into
```vue
<template #grid="slotProps">
	<div class="grid grid-nogutter">
		<div v-for="(data, index) in slotProps.items" :key="index" class="col-12 md:col-4">
			...		
		</div>
	</div>
</template>
```

but reading the changelog, it looks like there have been even more breaking changes since 3.38 (it seems this is the used version for the theme, according to the `package-lock.json`) - it is therefore possible that other pages do not work either.
The breaking changes can therefore also be corrected manually. However, it would be best if the Prime team updated the templates to the latest PrimeVue version, both for the checkbox issue and the breaking changes.

**Comment by ace-featherlight** [comment]
I use paid template [Avalon Vue - v5.2.0](https://avalon.primevue.org/) and have some issue @sezisfurkan
At the moment I am still in the development stage and thankfully it has not yet risen to production so I can wait for new version of your template, but for some other friends maybe this issue is quite disturbing

But I understand because the team (@cagataycivici @mertsincan @tugcekucukoglu and other) is now working on an update and refactoring for theming API and design in the latest version of prime, I believe this is indeed worth the wait and I'm very happy with the latest release of [primevue v.3.38](https://github.com/primefaces/primevue/releases/tag/3.48.0) So congratulations to the team for the latest version and its hard work ❤️ 

Regarding the issue that currently makes paid themes cannot run in local, I have some suggestions for the team

- Using `vite 5`, and all dependencies using the latest version `^`, because in the development stage it is impossible for us to use the strict version. because we still need every virtue and good of the latest version. We need the latest dependency version in every paid template version update
- Using the `Latest Theming API` methodology and technology currently used in Prime (Vue) Design
- Although maybe we can implement it ourselves, but there are many `New Components` that I hope can be included on templates
- I am a fan of your new theme which is `Aura`, I hope all paid template can be perfectly adopted and have the same design language of your all latest theme
- After I applied some form components in one page, I needed size options in all form components, buttons and other related things, not only in the one in `InputText` (even now I didn't find size option in the latest version)

Thank you

**Reply by lpalacecr** [reply]
I took on the task of trying primeblocks where I also bought a license and it also fails. (@tugcekucukoglu @cagataycivici  @mertsincan  and others) In my opinion, each update should take into account the impact it has on paid products, precisely for that reason, because they are paid, and should have priority. In my opinion, Primevue is one of the best tools I have used and I selected it after evaluating many, it would be a shame to have to stop using it due to these inconveniences.

Translated by Google Translate.

Original text:

Me di a la tarea de probar el primeblocks donde también compré una licencia y también falla. (@cagataycivici @mertsincan @tugcekucukoglu and other) En mi opinión cada actualización debería tomar en cuenta el impacto que tiene en los productos de paga, precisamente por eso, porque son de paga, y deberían tener prioridad. En mi criterio primevue es de las mejores herramientas que he utilizado y la seleccioné después de evaluar muchas, sería una lástima tener que dejar de usarla por estos inconvenientes.

**Reply by ace-featherlight** [reply]
I think we must move / create new discussion on template section @lpalacecr  , Until now there has been no answer from the team, I am afraid that the more out of the first page, this discussion will disappear

![image](https://github.com/primefaces/community/assets/78999773/2c8cdcbe-037d-477d-8572-a2bc2a6882ab)

**Reply by malacay1462** [reply]
Makes sense - in PrimeVue Templates is already another discussion: https://github.com/orgs/primefaces/discussions/1170
